vng_api_common/client.py

```diff
@@ -1,17 +1,67 @@
 """
-Interface to get a zds_client object for a given URL.
+Interface to get a client object for a given URL.
 """
 
-from typing import Optional
+import logging
+import time
+from typing import Any, Optional
 
-from django.apps import apps
 from django.conf import settings
 from django.utils.module_loading import import_string
 
-from zds_client import Client
+import jwt
+from ape_pie import APIClient
+from requests import JSONDecodeError, RequestException, Response
 
+JWT_ALG = "HS256"
 
-def get_client(url: str, url_is_api_root=False) -> Optional[Client]:
+
+logger = logging.getLogger(__name__)
+
+
+class ClientError(RuntimeError):
+    pass
+
+
+# TODO: use more approriate method name
+def to_internal_data(response: Response) -> dict | list | None:
+    try:
+        response_json = response.json()
+    except JSONDecodeError:
+        logger.exception("Unable to parse json from response")
+        response_json = None
+
+    try:
+        response.raise_for_status()
+    except RequestException as exc:
+        if response.status_code >= 500:
+            raise
+        raise ClientError(response_json if response_json is not None else {}) from exc
+
+    assert response_json
+    return response_json
+
+
+class Client(APIClient):
+    def request(
+        self, method: str | bytes, url: str | bytes, *args, **kwargs
+    ) -> Response:
+
+        headers = kwargs.pop("headers", {})
+        headers.setdefault("Accept", "application/json")
+        headers.setdefault("Content-Type", "application/json")
+
+        kwargs["headers"] = headers
+
+        data = kwargs.get("data", {})
+
+        if data:
+            kwargs["json"] = data
+
+        return super().request(method, url, *args, **kwargs)
+
+
+def get_client(url: str) -> Client | Any | None:
```

Comment:
Do we need this custom `get_client` with the `CLIENT_CLASS` setting? I think we should be able to use `build_client` from zgw_consumers together with the `ZGWAuth` class to create a client

Open Forms does this as well: https://github.com/open-formulieren/open-forms/blob/683efe79d912e977da79a6bddff5d077ceee28fd/src/openforms/registrations/contrib/zgw_apis/client.py#L23-L26

Reply:
We could remove the setting and use the `build_client` function. This will prevent applications to use their own client of course but that might be a considered decision

Reply:
Answered through https://github.com/maykinmedia/commonground-api-common/pull/38#discussion_r1816459670